src/cli/repo/filesystem.go

```diff
@@ -0,0 +1,215 @@
+		Long:    `Bootstraps a new repository on local or network storage and connects it to your m365 account.`,
+		RunE:    initFilesystemCmd,
+		Args:    cobra.NoArgs,
+		Example: fsProviderCmdInitExamples,
+	}
+}
+
+// initializes a filesystem repo.
+func initFilesystemCmd(cmd *cobra.Command, args []string) error {
+	ctx := cmd.Context()
+	overrides := flags.FilesystemFlagOverrides(cmd)
+
+	// TODO(pandeyabs): Move filepath conversion to FilesystemConfig scope.
+	abs, err := utils.MakeAbsoluteFilePath(overrides[flags.FilesystemPathFN])
+	if err != nil {
+		return Only(ctx, clues.Wrap(err, "getting absolute repo path"))
+	}
+
+	overrides[flags.FilesystemPathFN] = abs
+
+	cfg, err := config.GetConfigRepoDetails(
+		ctx,
+		storage.ProviderFilesystem,
+		true,
+		false,
+		flags.FilesystemFlagOverrides(cmd))
+	if err != nil {
+		return Only(ctx, err)
+	}
+
+	opt := utils.ControlWithConfig(cfg)
+	// Retention is not supported for filesystem repos.
+	retention := ctrlRepo.Retention{}
+
+	// SendStartCorsoEvent uses distict ID as tenant ID because repoID is still not generated
+	utils.SendStartCorsoEvent(
+		ctx,
+		cfg.Storage,
+		cfg.Account.ID(),
+		map[string]any{"command": "init repo"},
+		cfg.Account.ID(),
+		opt)
+
+	sc, err := cfg.Storage.StorageConfig()
+	if err != nil {
+		return Only(ctx, clues.Wrap(err, "Retrieving filesystem configuration"))
+	}
+
+	storageCfg := sc.(*storage.FilesystemConfig)
+
+	m365, err := cfg.Account.M365Config()
+	if err != nil {
+		return Only(ctx, clues.Wrap(err, "Failed to parse m365 account config"))
+	}
+
+	r, err := repository.Initialize(
+		ctx,
+		cfg.Account,
+		cfg.Storage,
+		opt,
+		retention)
+	if err != nil {
+		if flags.SucceedIfExistsFV && errors.Is(err, repository.ErrorRepoAlreadyExists) {
+			return nil
+		}
+
+		return Only(ctx, clues.Wrap(err, "Failed to initialize a new filesystem repository"))
+	}
+
+	defer utils.CloseRepo(ctx, r)
+
+	Infof(ctx, "Initialized a repository at path %s", storageCfg.Path)
+
+	if err = config.WriteRepoConfig(ctx, sc, m365, opt.Repo, r.GetID()); err != nil {
+		return Only(ctx, clues.Wrap(err, "Failed to write repository configuration"))
+	}
+
+	return nil
+}
+
+// ---------------------------------------------------------------------------------------------------------
+// Connect
+// ---------------------------------------------------------------------------------------------------------
+
+// `corso repo connect filesystem [<flag>...]`
+func filesystemConnectCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:     fsProviderCmd,
+		Short:   "Connect to a repository on local or network storage.",
+		Long:    `Ensures a connection to an existing repository on local or network storage.`,
+		RunE:    connectFilesystemCmd,
+		Args:    cobra.NoArgs,
+		Example: fsProviderCmdConnectExamples,
+	}
+}
+
+// connects to an existing filesystem repo.
+func connectFilesystemCmd(cmd *cobra.Command, args []string) error {
```

Comment:
For another PR: could I ask you to create generic `connectRepo` and `initRepo` handlers, much like we do with other cli commands?  Since azure should be a fast-follow here, it'll cut down on the boilerplate duplication for that implementation.